contracts/MatchPayouts.sol

```diff
@@ -0,0 +1,160 @@
+
+  /// @dev Address that can modify contract state
+  address public immutable owner;
+
+  /// @dev Address where funding comes from (Gitcoin Grants multisig)
+  address public immutable funder;
+
+  /// @dev Token used for match payouts
+  IERC20 public immutable dai;
+
+  /// @dev Convenience type used for setting inputs
+  struct PayoutFields {
+    address recipient; // grant receiving address
+    uint256 amount; // match amount for that grant
+  }
+
+  /// @dev Maps a grant's receiving address their match amount
+  mapping(address => uint256) public payouts;
+
+  /// @dev When true, the payouts mapping is finalized and cannot be updated
+  bool public finalized;
+
+  /// @dev When true, the contract has been funded and withdrawals can begin. Contract must be
+  /// `finalized` before it can be `funded`
+  bool public funded;
+
+  // =========================================== EVENTS ============================================
+
+  /// @dev Emitted when the `finalized` flag is set
+  event Finalized();
+
+  /// @dev Emitted when the `funded` flag is set
+  event Funded();
+
+  /// @dev Emitted when the funder reclaims the funds in this contract
+  event FundingWithdrawn();
+
+  /// @dev Emitted when a payout `amount` is added to the `recipient`'s payout total
+  event PayoutAdded(address recipient, uint256 amount);
+
+  /// @dev Emitted when a `recipient` withdraws their payout
+  event PayoutClaimed(address recipient, uint256 amount);
+
+  // ================================== CONSTRUCTOR AND MODIFIERS ==================================
+
+  /**
+   * @param _owner Address of contract owner
+   * @param _funder Address of funder
+   * @param _dai DAI address
+   */
+  constructor(address _owner, address _funder, IERC20 _dai) {
+    owner = _owner;
+    funder = _funder;
+    dai = _dai;
+  }
+
+  /// @dev Requires caller to be the owner
+  modifier onlyOwner() {
+    require(msg.sender == owner, "MatchPayouts: caller is not the owner");
+    _;
+  }
+
+  /// @dev Only allows method to be called once the payouts mapping has been finalized
+  modifier onlyWhenFinalized() {
+    require(finalized, "MatchPayouts: Payouts not finalized");
+    _;
+  }
+
+  // ======================================= PRIMARY METHODS =======================================
+  // Functions are laid out in the order they will be called over the lifetime of the contract
+
+  /**
+   * @notice Set's the mapping of addresses to their match amount
+   * @dev This will need to be called multiple times to prevent exceeding the block gas limit, based
+   * on the number of grants
+   * @param _payouts Array of `Payout`s to set
+   */
+  function setPayouts(PayoutFields[] calldata _payouts) external onlyOwner {
+    require(!finalized, "MatchPayouts: Payouts already finalized");
+    // Set each payout amount
+    for (uint256 i = 0; i < _payouts.length; i += 1) {
+      payouts[_payouts[i].recipient] = _payouts[i].amount;
```

Comment:
That is just a newbie question, but is it intended that the recipient amount could be overwritten by a subsequent call? If so, I would appreciate a comment there about that, thanks!

Reply:
Yep that is intended! Added some clarifying comments in 7da4d9a